pkg/pushsync/pushsync.go

```diff
@@ -633,12 +544,22 @@ func (ps *PushSync) pushToNeighbour(ctx context.Context, peer swarm.Address, ch
 	}
 	defer creditAction.Cleanup()
```

Comment:
could `replicateWithPeer` reuse `prepareCredit` function to prepare credit action?